nistats/first_level_model.py

```diff
@@ -557,3 +568,214 @@ def compute_contrast(self, contrast_def, stat_type=None,
+                                       file_tag='bold', file_type='json',
+                                       filters=filters)
+        if not img_specs:
+            warn('No preproc.json found in derivatives folder and no bold.json'
+                 ' in dataset folder. t_r can not be inferred and will need to'
+                 ' be set manually in the list of models, otherwise their fit '
+                 'will throw an exception')
+        else:
+            specs = json.load(open(img_specs[0], 'r'))
+            if 'RepetitionTime' in specs:
+                TR = float(specs['RepetitionTime'])
+            else:
+                warn('RepetitionTime not found in file %s. t_r can not be '
+                     'inferred and will need to be set manually in the '
+                     'list of models. Otherwise their fit will throw an '
+                     ' exception' % img_specs[0])
+            if 'SliceTimingRef' in specs:
+                SliceTimingRef = float(specs['SliceTimingRef'])
+            else:
+                warn('SliceTimingRef not found in file %s. It will be assumed'
+                     ' that the slice timing reference is 0.0 percent of the '
+                     'repetition time. If it is not the case it will need to '
+                     'be set manually in the generated list of models' %
+                     img_specs[0])
+
+    # Infer subjects in dataset
+    sub_folders = glob.glob(os.path.join(derivatives_path, 'sub-*'))
+    sub_labels = [os.path.basename(s).split('-')[1] for s in sub_folders]
+    sub_labels = sorted(list(set(sub_labels)))
+
+    # Build fit_kwargs dictionaries to pass to their respective models fit
+    # Events and confounds files must match number of imgs (runs)
+    models = []
+    models_fit_kwargs = []
+    for sub_label in sub_labels:
+        # Create model
+        model_kwargs = {'t_r': TR, 'slice_time_ref': SliceTimingRef,
+                        'subject_label': sub_label}
+        if model_init is not None:
+            model_kwargs.update(model_init)
+        model = FirstLevelModel(**model_kwargs)
+        models.append(model)
+
+        # Obtain model fit kwargs
+        model_fit_kwargs = {}
+
+        # Get preprocessed imgs
+        filters = [('task', task_id)]
+        # If preproc_variant is provided we add it to the search constraints
+        # for preproc imgs. The same for preproc_space.
+        if preproc_variant is not None:
+            filters.append(('variant', preproc_variant))
+        if preproc_space is not None:
+            filters.append(('space', preproc_space))
+
+        imgs = get_bids_files(derivatives_path, file_folder='func',
+                              file_tag='preproc', file_type='nii*',
+                              sub_label=sub_label, filters=filters)
+        # If there is more than one file for the same (ses, run), likely we
+        # have an issue of underspecification of filters.
```

Comment:
What do you mean by the _RL and _LR data? this would capture *preproc.nii files. Then, the idea I had here is that if there are many variants and spaces, since we can not really infer a default, I throw an error.